rpc/server.go

```diff
@@ -116,6 +110,64 @@ func marshalResponse(response any, log *zerolog.Logger) []byte {
 	return responseData
 }
 
+type validationResult struct {
+	Error  []byte
```

Comment:
Don't feel strongly, but curious about bundling the error and result together like this rather than returning a tuple `(result,error)` (which I am more used to seeing)?

Reply:
We could certainly return a tuple with an error. The reason the "error" is part of the struct is that it is not a golang error but a slice of bytes to be sent over to the requester.

Reply:
That makes sense!